From 4b946a9d4f9707b29086c9fcbdc6adcb288700be Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Fri, 28 Jun 2024 14:05:20 +0200
Subject: [PATCH 1004/1695] dts: stm32: add shadow provisionning of HCONF1 on
 stm32mp215f-dk board

Update the shadow default value to allow the system
to freeze the watchdog in stop or standby.
Do not write bit(16) to enable lower voltage on VDDIO1 because
SD Card is only alimented in 3.3V.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git da0217c989a53f9c7434f349a098a65da20428f7 dts: stm32: add shadow provisionning of HCONF1 on stm32mp215f-dk board]
Change-Id: If2b27f98154b22756ca4fb4448a9d22d6f8a5514
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390346
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index fc5b2c8a9..15b934535 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -31,6 +31,15 @@
 		device_type = "memory";
 		reg = <0x0 0x80000000 0x0 0x80000000>;
 	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+		hconf1_prov {
+			nvmem-cells = <&hconf1_otp>;
+			/* SDcard (SDMMC1) only in 3.3V */
+			st,shadow-value = <0x00008DB6>;
+		};
+	};
 };
 
 &cpu0 {
-- 
2.47.2

